pkg/controllers/raycluster_controller.go

```diff
@@ -226,6 +226,10 @@ func (r *RayClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{RequeueAfter: requeueTime}, err
 		}
 
+		if err := r.checkAndDeleteHeadPod(ctx, cluster); err != nil {
```

Comment:
I'd suggest to move it close to the ServiceAccount apply call to make it clear that it's related.